qiskit/dagcircuit/collect_blocks.py

```diff
@@ -54,7 +54,7 @@ def __init__(self, dag: DAGCircuit | DAGDependency):
 
         self.dag = dag
         self._pending_nodes: list[DAGOpNode | DAGDepNode] | None = None
-        self._in_degree: dict[DAGOpNode | DAGDepNode, int] | None = None
+        self._in_degree: dict[int, int] | None = None
```

Comment:
This is problematic, using the node ids is only safe for a single dag. If we want to switch this to be keyed on node ids we can **not** cache it. We could get away with the cache with the `DAGNode` storage because the hash and eq implementations are unique; either pre-rust dagcircuit where it was object id based or post-rust dag where this is an actual implementation. But node ids are not unique between dags and the indices are reused between dags and will refer to different nodes.

Reply:
I see the point, but a single `BlockCollector` class only makes sense for a single DAG, in which case we should be fine using node ids, correct?